resources/windows/wix/customactions.wxs.erb

```diff
@@ -250,6 +250,7 @@ If (fso.DirectoryExists(dataDirectory)) Then
   ' ensure admins and system have full control
   ExecuteCommand comspec & " /C " & icacls & "  """ & dataDirectory & "\*"" /grant ""*S-1-5-32-544:(F)"" ""*S-1-5-18:(F)"" /T /C 2>&1"
 
+  ExecuteCommand comspec & " /C " & icacls & "  """ & dataDirectory & "\*"" /reset /T /C 2>&1"
```

Comment:
I'm about dubious about this.  First we set perms and then reset them back to inherited only.

Reply:
I'll double check that I didn't transcribe anything incorrectly here, but it should be an exact port of the existing custom actions.

Reply:
I can understand `reset then set` but not the other way around.

Reply:
This ordering matches what was there already from @joshcooper and the sequence of events makes sense to me:

* take ownership - `TakeOwnPuppetPermissions`
* set parent directory to Administrators (F) and SYSTEM (F) (recursively) - `GrantPuppetPermissions`
* reset parent recursively - `ResetPuppetPermissions`

Note what reset does:

```
ICACLS name /reset [/T] [/C] [/L] [/Q]
    replaces ACLs with default inherited ACLs for all matching files.
```

I believe Josh tested this extensively and found that the reset was necessary to get the inheritance properly established.


Only difference in the end here is that code, facter, pxp-agent were previously reset *before* `puppet` directory. But given they are siblings that doesn't matter.